### PR TITLE
Sign up form validation test

### PR DESCRIPTION
**Pull request by shreyaa-sharmaa**
Signup form validation test

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Review by jywarren — APPROVED** [review]

File: test/system/signup_form_test.rb [hunk 1]
```diff
@@ -1,4 +1,21 @@
...
 class SignupFormTest < ApplicationSystemTestCase
+    def setup
+      visit "/"
+    end
+  
+    test "the signup form is validated" do 
+        visit "/signup" 
+       
+        assert_difference 'User.count', 0 do
+            #Signs up with registered email
```
Comment (jywarren) [inline test/system/signup_form_test.rb:12]:
```suggestion
            # Signs up with an already-registered email 
```

**Comment by jywarren** [comment]
Just a tiny rewording to clarify the test!

File: test/system/signup_form_test.rb [hunk 2]
```diff
@@ -1,4 +1,21 @@
 require "application_system_test_case"
 
 class SignupFormTest < ApplicationSystemTestCase
+    def setup
+      visit "/"
+    end
+  
+    test "the signup form is validated" do
```
Comment (jywarren) [inline test/system/signup_form_test.rb:8]:
```suggestion
    test "the signup form is validated and rejects a duplicate email" do 
```

**Comment by shreyaa-sharmaa** [comment]
I'm adding a test in this for a duplicate username as well.

**Comment by shreyaa-sharmaa** [comment]
#8469 is failing. Have we made the corresponding changes in the code? For the user not to register if the spam test is not passed? 

I checked and they fail locally too. Should I open a PR to revert it?

**Comment by jywarren** [comment]
Whoa that is so strange... look, it totally passed when I merged it!

image

What do you see is the error now? can u link me? I think we don't have to revert...

**Comment by shreyaa-sharmaa** [comment]
https://travis-ci.com/github/publiclab/plots2/jobs/396092060#L2389

**Comment by jywarren** [comment]
Hmm whoaaaa. Such inconsistency! 

```
 FAIL["test_the_user_is_not_registered_without_passing_spam_", #<Minitest::Reporters::Suite:0x00007f20b1371c38 @name="SignupFormTest">, 104.59803336399997]
 test_the_user_is_not_registered_without_passing_spam_#SignupFormTest (104.60s)
        "User.count" didn't change by 0.
        Expected: 19
          Actual: 20
        test/system/signup_spamaway_test.rb:11:in `block in <class:SignupFormTest>'
```

Um, i'm working on getting https://github.com/publiclab/plots2/pull/8467 finished, so why don't we just note that that should solve the problem and then rebase this... and then address the inconsistency later. All of this can happen later though! 

Perhaps we should add something after https://github.com/publiclab/plots2/pull/8469/files#diff-e0e966662d6f0b4c06e82346b2605588R18 to ensure that it completes -- the clicking of the signup button doesn't necessarily confirm that the subsequent page has loaded. 

For reference, adding a css-based assertion after a click event can help to confirm that the page has indeed reloaded fully: 

https://github.com/publiclab/plots2/blob/eab9696cecf55d696d7f7844539ae3bf46acf001/test/system/comment_test.rb#L103-L110

**Comment by jywarren** [comment]
Could we add such a css selector to the end of the system test, in this PR, to both system tests, to try to resolve both?

**Comment by shreyaa-sharmaa** [comment]
Yeah sure. If it's not urgent can I get back to it tomorrow? I have a class to attend in 4 hours :sweat_smile:

**Comment by jywarren** [comment]
Oh my gosh Shreya, please go to sleep!!! Go!!! Really it's so important! ❤️

**Comment by jywarren** [comment]
Forgive me, i know i said so but since you wanted to keep going i kept replying. Let's resume this another time! Many thanks!

**Comment by shreyaa-sharmaa** [comment]
No worries. I felt responsible for my code. Adios!

**Comment by jywarren** [comment]
It'll be here when you come back 😄 

thanks again!

**Comment by jywarren** [comment]
OK -- https://github.com/publiclab/plots2/pull/8467 is merged! Let's rebase this over to see.

**Comment by jywarren** [comment]
Done here! Thank you!!!!
